napari/layers/shapes/shapes.py

```diff
@@ -474,6 +474,7 @@ def __init__(
             if ndim is None:
                 ndim = 2
             data = np.empty((0, 0, ndim))
+            warmup_numba_cache()
```

Comment:
```suggestion
            if not get_settings()._experimental.compiled_triangulation:
                warmup_numba_cache()
```

Reply:
I do not want to add settings to next place. I need to rethink how to do this.

Reply:
This was resolved because `warmup_numba_cache` checks whether the compiled backend is in use internally.